commit df0c32c987bb5d340fb18e817b312c3e3da4e215
Author: Luiz Augusto von Dentz <luiz.dentz-von@nokia.com>
Date:   Tue Jul 27 15:22:46 2010 +0300

    core: fix reading adapter name on DEVUP
    
    Reading the name on adapter_start is useless since adapter_setup will
    just overwrite it with the name stored in the filesystem.

diff --git a/src/adapter.c b/src/adapter.c
index 789a196..ad0ac91 100644
--- a/src/adapter.c
+++ b/src/adapter.c
@@ -2351,8 +2351,6 @@ int adapter_start(struct btd_adapter *adapter)
 
 	memcpy(dev->features, features, 8);
 
-	adapter_ops->read_name(adapter->dev_id);
-
 	if (!(features[6] & LMP_SIMPLE_PAIR))
 		goto setup;
 
From 80536745f90f73404b3d3507ea81b35498b2e7e3 Mon Sep 17 00:00:00 2001
From: Luiz Augusto von Dentz <luiz.dentz-von@nokia.com>
Date: Thu, 29 Jul 2010 15:01:19 +0300
Subject: [PATCH] core: fix not emitting 'Name' PropertyChanged when adapter is down

---
 src/adapter.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/adapter.c b/src/adapter.c
index ad0ac91..4615326 100644
--- a/src/adapter.c
+++ b/src/adapter.c
@@ -993,6 +993,9 @@ static DBusMessage *set_name(DBusConnection *conn, DBusMessage *msg,
 	if (!adapter->up) {
 		strncpy((char *) adapter->dev.name, name, MAX_NAME_LENGTH);
 		write_local_name(&adapter->bdaddr, name);
+		emit_property_changed(connection, adapter->path,
+					ADAPTER_INTERFACE, "Name",
+					DBUS_TYPE_STRING, &name);
 	} else {
 		int err = adapter_ops->set_name(adapter->dev_id, name);
 		if (err < 0)
-- 
1.7.0.4

